### ISSUE TITLE
ML:Implement Feature Scaling by Selecting Best Scaler per Column and Save to New CSV

### ISSUE THREAD

**Issue by Nidhi-Satyapriya** [Nidhi-Satyapriya/AutoEDA-Automated-Data-Preprocessing-Toolkit#23]
###  Overview

We want to automate **feature scaling** in autoEDA by testing **multiple scalers** on each numeric column and selecting the one that gives the most normalized output. This will `improve` **model performance** and ensure consistency.

Your task is to:
1. Apply 3 `scalers `to each numeric column: `StandardScaler`, `MinMaxScaler`, and `RobustScaler`.
2. Measure how well each performs (based on **skewness** — closer to 0 is better).
3. Select the best-performing scaler **for each column individually**.
4. Transform the data accordingly.
5. Save the entire new scaled DataFrame to a `CSV file` (`autoEDA_scaled_output.csv`).
6. Return a report showing which scaler was used for each column.

---

### ✅ Goals

- [ ] Detect numeric columns (exclude binary and ID-like features).
- [ ] Apply all 3 scalers to each column and calculate skewness.
- [ ] Select and apply the best scaler per column.
- [ ] Create a final DataFrame with all selected-scaled columns.
- [ ] Save the result as `autoEDA_scaled_output.csv`.
- [ ] Return a dictionary report: `{ column_name: scaler_used }`.

---

###  Tools to Use

- `pandas`
- `sklearn.preprocessing`: `StandardScaler`, `MinMaxScaler`, `RobustScaler`
- `scipy.stats.skew` for measuring normalization performance
- `numpy` for transformations

---

###  Example

```python
scaled_df, scaler_report = auto_scale_features(df)
scaled_df.to_csv("autoEDA_scaled_output.csv", index=False)
print(scaler_report)
# Output: { "age": "StandardScaler", "income": "RobustScaler" }

### Instructions:
1) Put the sample dataset used in 'notebooks/sample_csv'.
2) Prepare and `document `it in a Notebook.
3) Keep the steps `generalized `for any csv.

**Comment by Sparsh2105** [comment]
And which dataset do i have to use?

**Comment by Sparsh2105** [comment]
> And which dataset do i have to use? 

@Nidhi-Satyapriya please clarify

**Comment by Nidhi-Satyapriya** [comment]
Assigned! @Sparsh2105.
you are free to choose any, remember to put it in sample_csv directory.
Prefer titanic since one contributor is using that.
Make sure to cover edge cases of preprocessing, even though not required in your dataset, via making a function, so that it generalizes to common situations as well.

**Comment by Sparsh2105** [comment]
I will make my PR in 3 days kindly grant me this much time.

**Comment by Sparsh2105** [comment]
Where should i keep my Collab notebook containing the code and analysis in my PR > i mena which directory ?

**Comment by Shivu384** [comment]
Hi! @Nidhi-Satyapriya  I see this issue is already assigned — just checking if there's still work being done on it. I'd love to contribute if it's available. Let me know.

**Comment by Nidhi-Satyapriya** [comment]
@Sparsh2105 PR in `/notebooks  `directory at your earliest convenience.
